plugins/cpu/test_cpu_plugin.cc

```diff
@@ -0,0 +1,25 @@
+#include "plugins/cpu/test_cpu_plugin.h"
+
+#include <iostream>
+
+#include "xla/pjrt/c/pjrt_c_api.h"
+#include "xla/pjrt/c/pjrt_c_api_cpu_internal.h"
+
+// Use `test` as the platform name instead of `cpu` so torch_xla treats this
+// as an unknown device.
+PJRT_Error* test_platform_name(PJRT_Client_PlatformName_Args* args) {
+  static const std::string platform_name = "test";
+  args->platform_name = platform_name.c_str();
+  args->platform_name_size = platform_name.size();
+  return nullptr;
+}
+
+const PJRT_Api* GetPjrtApi() {
+  // HACK: The CPU client is created as a constexpr, so const-casting is
+  // undefined behavior. Make a non-const copy of the struct so we can override
```

Comment:
Wondering where you "Make a non-const copy of the struct"

Reply:
`pjrt_api` is a value type, so assignment is implicitly a copy.